sos/policies/distros/redhat.py

```diff
@@ -298,7 +298,7 @@ def get_upload_url(self):
             self.ui_log.info("No case id provided, uploading to SFTP")
             return RH_SFTP_HOST
         rh_case_api = "/support/v1/cases/%s/attachments"
-        return RH_API_HOST + rh_case_api % self.case_id
+        return RH_API_HOST + rh_case_api % self.commons['cmdlineopts'].case_id
```

Comment:
Why this change? AFAIK `self.case_id` might be blank (and common's `case_id` set) only in scenario "case id not in cmdline, batch not in cmdline" - should not upload query for case_id, then? (or am I wrong here with my assumption)?

(that concern is valid for sure (while I can be wrong on its impact to this code change):

    # python3 bin/sos upload /var/tmp/sosreport-pmoravec-rhel8-012345678-2024-08-13-gbiatgg.tar.xz
    
    sos upload (version 4.7.2)
    This utility is used to upload files to a policy-default location.
    
    The archive to be uploaded may contain data considered sensitive and its content
    should be reviewed by the originating organization before being passed to any
    third party.
    
    No configuration changes will be made to the system running this utility.
    
    
    Press ENTER to continue, or CTRL-C to quit
    
    
    Attempting to upload file /var/tmp/sosreport-pmoravec-rhel8-012345678-2024-08-13-gbiatgg.tar.xz to case 
    No case id provided, uploading to SFTP
    No case id provided, uploading to SFTP
    Attempting upload to Red Hat Secure FTP
    ..

Reply:
This is one of the things we talked about internally when I first started playing with 'upload'. If you remember, the issue was that without this change, we were getting 'None' on the case_id and it was failing to build the url, and so failed to upload. I have the feeling that I've done something wrong on the upload side and I'm not passing the case_id correctly.
My hope is that more experienced eyes, or at least fresher, can tell me where I'm failing.